templates/typescript_gapic/_util.njk

```diff
@@ -9,3 +9,57 @@
             {{ chain }}.{{ method.headerRequestParams.slice(-1)[0] }} = '';
 {%- endif %}
 {%- endmacro -%}
+
+{%- macro printCommentsForParams(method) -%}
+{%- set commentsMap = method.paramComment -%}
+{%- for oneComment in commentsMap -%}
+{%- set type = oneComment.paramType -%}
+{%- if type.startsWith('.') %}
+ * @param { {{- type.substring(1) -}} } request.{{ oneComment.paramName }}
+{%- else %}
+ * @param { {{- convertParamType(oneComment.paramType) -}} } request.{{ oneComment.paramName }}
+{%- endif -%}
+{%- set lines = oneComment.comments -%}
+{%- for line in lines %} 
+ *  {{ line.replaceAll('*/', '* /') | safe}} 
+{%- endfor -%}
+{%- endfor -%}
+{%- endmacro -%}
+
+{%- macro printCommentsForMethod(method) -%}
+{%- set lines = method.comments -%}
+{% for line in lines %}
+ *{{ line | safe}}
+{%- endfor %}
+{%- endmacro -%}
+
+{%- macro printCommentsForService(service) -%}
+{%- set lines = service.comments -%}
+{% for line in lines %}
+   * {{ line | safe}}
+{%- endfor %}
+{%- endmacro -%}
+
+{%- macro printRequest() %}
+ * @param {Object} request
+ *   The request object that will be sent.
+{%- endmacro -%}
+
+{%- macro printReturn() %}
+ * @returns {Promise} - The promise which resolves to an array.
+ *   The first element of the array is an object representing [LogMetric]{@link google.logging.v2.LogMetric}.
```

Comment:
Oh no! There should not be any `LogMetric` here, it should take a parameter.

Also, please notice that different type of calls (unary, streaming, paging, long running) have different `@return` descriptions. It's OK to have separate macros for all of them, or do `if longrunning ... else if paging ...` etc.

Reply:
I think this is the last comment here, everything else looks great!

Reply:
Thanks! Checked gapic-gen seems for @return part, there's also returnTypeName & return leading comments logic for printing, just like parameters. It seems not an easy fix, but let me explore more. :)